strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/gremlin/adapters/ArtifactAdapter.java

```diff
@@ -0,0 +1,246 @@
+    {
+        UnfoldEntityTraversal<Vertex, Vertex> artifactCoordinatesUnfold = artifactCoordinatesAdapter.unfold(artifactCoordinates);
+
+        return __.<S2>V(artifactCoordinates)
+                 .saveV(artifactCoordinates.getUuid(),
+                        artifactCoordinatesUnfold);
+    }
+
+    private EntityTraversal<Vertex, Vertex> unfoldArtifact(Artifact entity)
+    {
+        EntityTraversal<Vertex, Vertex> t = __.<Vertex>identity();
+
+        if (entity.getStorageId() != null)
+        {
+            t = t.property(single, "storageId", entity.getStorageId());
+        }
+        if (entity.getRepositoryId() != null)
+        {
+            t = t.property(single, "repositoryId", entity.getRepositoryId());
+        }
+        if (entity.getCreated() != null)
+        {
+            t = t.property(single, "created", toLong(entity.getCreated()));
+        }
+        if (entity.getLastUpdated() != null)
+        {
+            t = t.property(single, "lastUpdated", toLong(entity.getLastUpdated()));
+        }
+        if (entity.getLastUsed() != null)
+        {
+            t = t.property(single, "lastUsed", toLong(entity.getLastUsed()));
+        }
+        if (entity.getSizeInBytes() != null)
+        {
+            t = t.property(single, "sizeInBytes", entity.getSizeInBytes());
+        }
+        if (entity.getDownloadCount() != null)
+        {
+            t = t.property(single, "downloadCount", entity.getDownloadCount());
+        }
+
+        ArtifactArchiveListing artifactArchiveListing = entity.getArtifactArchiveListing();
+
+        Set<String> filenames = artifactArchiveListing.getFilenames();
+        t = t.sideEffect(__.properties("filenames").drop());
+        t = t.property("filenames", filenames);
+
+        Map<String, String> checksums = entity.getChecksums();
+        Set<String> checkSumAlgo = new HashSet<>();
+        for (String alg : checksums.keySet())
+        {
+            checkSumAlgo.add("{" + alg + "}" + checksums.get(alg));
+        }
+        t = t.sideEffect(__.properties("checksums").drop());
+        t = t.property("checksums", checkSumAlgo);
+
+        if (entity.getArtifactFileExists() != null)
+        {
+            t = t.property(single, "artifactFileExists", entity.getArtifactFileExists());
+        }
+
+        return t;
+    }
+
+    @Override
+    public EntityTraversal<Vertex, Element> cascade()
+    {
+        return __.<Vertex>aggregate("x")
+                //TODO: remove ArtifactAoordinates
+//                 .optional(__.outE(Edges.ARTIFACT_HAS_ARTIFACT_COORDINATES)
+//                             .inV()
+//                             .where(__.inE(Edges.ARTIFACT_HAS_ARTIFACT_COORDINATES).count().is(1))
+//                             .aggregate("x")
+//                             .inE(Edges.ARTIFACT_COORDINATES_INHERIT_GENERIC_ARTIFACT_COORDINATES)
+//                             .outV()
+//                             .aggregate("x"))
```

Comment:
@sbespalov : Is this dead code? If so, could you please remove it? Thanks! :)

Reply:
I expected to get back to it when will have a time

Reply:
finally commented code and `TODO` removed